### ISSUE TITLE
How to add constraits (e.g. value region) of prediction in training? 

### ISSUE THREAD

**Issue by dou3516** [Microsoft/LightGBM#1184]
Please search your question on previous issues, [stackoverflow](https://stackoverflow.com/questions/tagged/lightgbm) or other search engines before you open a new one.

For bugs and unexpected issues, please provide following information, so that we could reproduce on our system.

## Environment info
Operating System:
CPU:
C++/Python/R version:

## Steps to reproduce

1.
2.
3.

**Comment by guolinke** [comment]
I think it isn't a standard function in boosted tree algorithm. 
Limiting the value of single tree is easy, you can just change this line(https://github.com/Microsoft/LightGBM/blob/master/include/LightGBM/tree.h#L14). 
However, the final prediction is based on all trees. 

One possible solution is using custom objective function with self-defined `ConvertOutput` (like https://github.com/Microsoft/LightGBM/blob/master/src/objective/binary_objective.hpp#L122-L124). But you will need to change the calculation of gradients and hessians accordingly.
